src/http/websocket_http_client.zig

```diff
@@ -363,7 +365,8 @@ pub fn NewHTTPUpgradeClient(comptime ssl: bool) type {
             this.cancel();
         }
 
-        pub fn handleClose(this: *HTTPClient, _: Socket, _: c_int, _: ?*anyopaque) void {
+        pub fn handleClose(this: *HTTPClient, socket: Socket, _: c_int, _: ?*anyopaque) void {
+            if (this.isDeadPtr()) return;
```

Comment:
This check should be moved up before we call the method on HTTPClient. It’s not an HTTPClient pointer here. It’s going to cause undefined behavior and crashes in release builds